ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java

```diff
@@ -240,8 +237,9 @@ StructuredDataSource getResultDatasource(final Select select, final String name)
     );
   }
 
-  private AbstractStreamCreateStatement maybeAddFieldsFromSchemaRegistry(
-      AbstractStreamCreateStatement streamCreateStatement
+  private StatementWithInferredSchema maybeAddFieldsFromSchemaRegistry(
+      AbstractStreamCreateStatement streamCreateStatement,
```

Comment:
can we make this `final` and not re-assign it in the method? i.e., return from line 269